### PR TITLE
[proofs] [alethe] Proper support for Skolems 

### LINKED ISSUE #9515: Seg fault in AletheProofPrinter
**Describe the bug**

On the following  file, cvc5 suffers a segfault when trying to produce a proof  in the Alethe format:
```
(set-logic QF_UF)
(declare-sort u 0)
(declare-fun $2 () Bool)
(declare-fun y () u)
(declare-fun $ () u)
(assert (= y (ite $2 y $)))
(assert (not $2))
(assert (not (= $ y)))
(check-sat)
```

This happens with a recent build: commit 6a0abb70032db7bd747724477ca63372e7c801eb.

To reproduce:
```
cvc5 --dump-proofs --proof-format-mode=alethe --dag-thresh=0 --simplification=none test.smt2
```
This should produce:
```
cvc5 suffered a segfault.
Offending address is 0x2f
Looks like a NULL pointer was dereferenced.
Segmentation fault
```

### PR DESCRIPTION
**Pull request by HanielB**
Have Alethe node converter properly handle skolems. Also adds the generation of the appropriate choice term for arrays extensionality. 

Fixes #9515 and #9516, and adds the reported problems as regression tests.

### CODE REVIEW

File: src/proof/alethe/alethe_node_converter.cpp [hunk 1]
```diff
@@ -21,39 +21,61 @@
...
+                           << "\n";
+      Node wi = SkolemManager::getOriginalForm(n);
+      Trace("alethe-conv") << "AletheNodeConverter: ..original: " << wi << "\n";
+      if (wi == n)
+      {
+        // if it is not a purification skolem, maybe it has a witness skolem
+        wi = SkolemManager::getWitnessForm(n);
+      }
+      // skolem with witness form, just convert that
+      if (!wi.isNull() && wi != n)
```
Comment (Lachnitt) [inline src/proof/alethe/alethe_node_converter.cpp:43]: if you use "else if" you can loose the wi != n condition

Comment (HanielB) [reply src/proof/alethe/alethe_node_converter.cpp:43]: Hum, the call in line 40 can modify `wi`, so using an `else if` and keep the same behavior will depend on what `getWitnessForm` is doing, i.e., if whenever it returns a non-null term it is not the same as `n`. This should always be the case, right @ajreynol ?

Comment (ajreynol) [reply src/proof/alethe/alethe_node_converter.cpp:43]: Yes, this is correct.

Although note that `getWitnessForm` may be eliminated in the near future.

Comment (HanielB) [reply src/proof/alethe/alethe_node_converter.cpp:43]: OK, thanks! And about eliminating `getWitnessForm`, well, as long as there is an equivalent functionality...

Comment (ajreynol) [reply src/proof/alethe/alethe_node_converter.cpp:43]: It will be another skolem function identifier `QUANTIFIERS_SKOLEMIZE`.

Comment (ajreynol) [reply src/proof/alethe/alethe_node_converter.cpp:43]: See https://github.com/cvc5/cvc5/pull/9536.

Comment (HanielB) [reply src/proof/alethe/alethe_node_converter.cpp]: Given the big change that #9536 introduces I'd like to merge this in first, as it fixes the aforementioned issues. And also provides a use case for skolems that #9536 must not break.

Comment (ajreynol) [reply src/proof/alethe/alethe_node_converter.cpp]: Sure, then you can formalize the skolem as a choice term here in whatever way you want.

Comment (HanielB) [reply src/proof/alethe/alethe_node_converter.cpp]: I have removed the thing I was playing around for array extensionality. The handling for quantifier skolemization however must be that one (or something equivalent to it) in Alethe's current form.

**Review by Lachnitt — APPROVED** [review]

**Pull request merged**